sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,186 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+import { CASSupplier, SupplierAddress, SupplierStatus } from "@sims/sims-db";
+import { ProcessSummary } from "@sims/utilities/logger";
+import {
+  CASAuthDetails,
+  CASSupplierResponseItem,
+} from "../../processors/schedulers/cas-integration/models/cas-supplier-response.dto";
+import { Not, Repository, UpdateResult } from "typeorm";
+import { CASIntegrationConfig, ConfigService } from "@sims/utilities/config";
+import { CASService } from "./cas.service";
+
+const CAS_SUPPLIER_ADDRESS_ACTIVE_STATUS = "ACTIVE";
+
+@Injectable()
+export class CASSupplierIntegrationService {
+  private readonly casIntegrationConfig: CASIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly casService: CASService,
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    this.casIntegrationConfig = config.casIntegration;
+  }
+
+  /**
+   * CAS integration process.
+   * Logs on CAS supplier API and request the supplier information for the students with pending supplier information.
+   * @param parentProcessSummary parent process summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @returns a number of update records.
+   */
+  async executeCASIntegrationProcess(
+    parentProcessSummary: ProcessSummary,
+    casSuppliers: CASSupplier[],
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    const summary = new ProcessSummary();
+    parentProcessSummary.children(summary);
+    try {
+      summary.info("Logging on CAS...");
+      const auth = await this.casService.casLogon();
+      if (auth.access_token) {
+        summary.info("Logon successful.");
+        suppliersUpdated = await this.requestCASAndUpdateSuppliers(
+          casSuppliers,
+          summary,
+          auth,
+        );
+      } else {
+        summary.info("Could not authenticate on CAS.");
```

Comment:
A failure to authenticate on CAS should be treated as an error inside the method `casLogon`. If the HTTP response is different than 200 an exception can be thrown. Same for `requestCASAndUpdateSuppliers`.
For  `requestCASAndUpdateSuppliers` a 404 is expected when the student is not found and it should be considered an expected response where the method should return `undefined` or something that would indicate it.

Reply:
Maybe we should throw a custom-named error when CAS auth fails to allow its detection in the processor and rethrown it to force the queue to retry the operation. Does it make sense?

Reply:
Maybe I was not clear in my comments but I need a confirmation, when the authentication fails the HTTP response will just be returned as empty?

Reply:
I changed that to check the exception instead.